include/pybind11/cast.h

```diff
@@ -495,13 +495,20 @@ template <typename T> typename make_caster<T>::template cast_op_type<T> cast_op(
     return cast_op<T>(caster);
 }
 
-template <typename type> class type_caster<std::reference_wrapper<type>> : public type_caster_base<type> {
+template <typename type> class type_caster<std::reference_wrapper<type>> {
+private:
+    using caster_t = make_caster<type>;
+    caster_t subcaster;
 public:
+    bool load(handle src, bool convert) { return subcaster.load(src, convert); }
+    static PYBIND11_DESCR name() { return caster_t::name(); }
     static handle cast(const std::reference_wrapper<type> &src, return_value_policy policy, handle parent) {
-        return type_caster_base<type>::cast(&src.get(), policy, parent);
+        return caster_t::cast(&src.get(), policy, parent);
```

Comment:
`copy` is the default currently if you `py::cast(x)` a `reference_wrapper<T> x`.  Essentially both `copy` and `reference` currently work the way I'd expect:

```C++
#include <pybind11/pybind11.h>

namespace py = pybind11;

struct Foo { int value = 2; };

std::vector<std::reference_wrapper<Foo>> foos() {
    static Foo foo1, foo2;
    std::vector<std::reference_wrapper<Foo>> r;
    r.emplace_back(foo1);
    r.emplace_back(foo2);
    return r;
}

PYBIND11_PLUGIN(copy) {
    py::module m("copy");

    py::class_<Foo>(m, "Foo").def(py::init<>()).def_readwrite("value", &Foo::value);

    m.def("z1", []() {
        py::list l;
        for (auto &f : foos()) l.append(py::cast(f, py::return_value_policy::copy));
        return l;
    });
    m.def("z2", []() {
        py::list l;
        for (auto &f : foos()) l.append(py::cast(f, py::return_value_policy::reference));
        return l;
    });

    return m.ptr();
}
```

```pycon
>>> from copy import z1, z2, Foo
>>> a = z1()
>>> a2 = z1()
>>> b = z2()
>>> b2 = z2()
>>> a[0] is a2[0]
False
>>> b[0] is b2[0]
True
```

(The `::copy` there is redundant: it'll be the default for an lvalue reference wrapper already.)